### PR TITLE
Perform search when hashtag is pressed

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
When a hashtag is pressed it takes the user to the global search screen an immediately performs a search of the hashtag.

```release-note
NONE
```

### CODE REVIEW

File: app/components/markdown/markdown.tsx [hunk 1]
```diff
@@ -288,6 +288,7 @@ const Markdown = ({
 
         return (
             <Hashtag
+                channelId={channelId}
```
Comment (Willyfrog) [inline app/components/markdown/markdown.tsx:291]: Sorry for my ignorance, but will this work properly in the search and recent mentions tabs?
Given the right context it should, but just confirming.

**Review by Willyfrog — APPROVED** [review]
Added a comment, a double check, but if that's ok it is LGTM

**Review by jfrerich — COMMENTED** [review]
When a user is already in the search tab and clicks on a search result post that contains a hashtag, should we handle that case differenctly than emitting a navigation to the search screen?  

I haven't tested this PR locally, but curious if this device emitter event it flickering the entire search tab?

File: app/components/markdown/markdown.tsx (hunk 1 above)
Comment (enahum) [reply app/components/markdown/markdown.tsx:291]: yes, see https://github.com/mattermost/mattermost-mobile/blob/main/app/components/post_list/post/body/message/message.tsx#L88

But I'm going to double check the global threads

Comment (enahum) [reply app/components/markdown/markdown.tsx:291]: ok, global threads do not have the hashtag tappable

**Comment by enahum** [comment]
> When a user is already in the search tab and clicks on a search result post that contains a hashtag, should we handle that case differenctly than emitting a navigation to the search screen?
> 
> I haven't tested this PR locally, but curious if this device emitter event it flickering the entire search tab?

you can test it, it'll work just the same and you won't notice anything navigation related.

**Review by esethna — CHANGES_REQUESTED** [review]
> performs a search of the hashtag in the channel the post belonged to.

To match webapp behaviour, i'd propose we run a search for the hashtag globally (ie the entire team, not filtered to the channel)

**Comment by jfrerich** [comment]
> > When a user is already in the search tab and clicks on a search result post that contains a hashtag, should we handle that case differenctly than emitting a navigation to the search screen?
> > I haven't tested this PR locally, but curious if this device emitter event it flickering the entire search tab?
> 
> @jfrerich
> 
> you can test it, it'll work just the same and you won't notice anything navigation related.
Sounds great.

**Review by jfrerich — APPROVED** [review]

**Comment by enahum** [comment]
> > performs a search of the hashtag in the channel the post belonged to.
> 
> To match webapp behaviour, i'd propose we run a search for the hashtag globally (ie the entire team, not filtered to the channel)

@esethna I think your proposal is taking the user out of the context of the channel were you want to search for other hashtags which matches the behavior on the webapp when you click the hashtag.

**Comment by esethna** [comment]
>  I think your proposal is taking the user out of the context of the channel

@enahum I think that's the intention of a hashtag search. ie it finds everything related to that topic, not necessarily isolated to the channel where the hashtag was posted. 

ie on webapp when i click a hashtag it executes a search for `#hashtag` not `#hashtag in:channel-name`

cc// @matthewbirtch if you want to weigh in here

**Comment by enahum** [comment]
@esethna I just checked on the webapp and clicking on a hashtag does search cross channel.. will make the change

**Review by esethna — APPROVED** [review]

**Pull request merged**